File: targets/java-spring/services/ToolService.java
package com.baddemo.services;

import com.baddemo.utils.ExecUtil;

/** Diagnostic command logic. */
public class ToolService {

    /** Value staged by the controller. */
    private String target;

    public void stageTarget(String host) {
        this.target = host;
    }

    /** Reads the staged field and runs the diagnostic. */
    public int runStagedDiag() throws Exception {
        return ExecUtil.run("ping -c 1 " + this.target);
    }
}

File: targets/php-laravel/app/Services/ToolService.php
<?php

namespace App\Services;

class ToolService
{
    private string $target = '';

    public function stageTarget(string $host): void
    {
        $this->target = $host;
    }

    public function runStagedDiag(): ?string
    {
        $cmd = "ping -c 1 " . $this->target;
        return shell_exec($cmd);
    }
}
